### PR TITLE
Add override option for app titles, #120

### PR DESCRIPTION
**Pull request by exiam**
Hello,

As I needed this feature, I added the possibility to override app titles.

You can use the new `appTitles` argument to pass an object with app code as key and custom title as value :

```js
showLocation({
  latitude: 38.8976763,
  longitude: -77.0387185,
  title: 'The White House', // optional
  dialogTitle: 'This is the dialog Title', // optional (default: 'Open in Maps')
  dialogMessage: 'This is the amazing dialog Message', // optional (default: 'What app would you like to use?')
  cancelText: 'This is the cancel button text', // optional (default: 'Cancel')
  appTitles: { 'google-maps': 'My custom Google Maps title' }
});
```

With Popup:

```ts
import { Popup } from 'react-native-map-link';

<Popup
    isVisible={this.state.isVisible}
    onCancelPressed={() => this.setState({ isVisible: false })}
    onAppPressed={() => this.setState({ isVisible: false })}
    onBackButtonPressed={() => this.setState({ isVisible: false })}
    modalProps={{ // you can put all react-native-modal props inside.
        animationIn: 'slideInUp'
    }}
    appsWhiteList={[ /* Array of apps (apple-maps, google-maps, etc...) that you want
    to show in the popup, if is undefined or an empty array it will show all supported apps installed on device.*/ ]}
    appTitles={{ waze: 'My custom Waze title' }}
    options={{ /* See `showLocation` method above, this accepts the same options. */ }}
    style={{ /* Optional: you can override default style by passing your values. */ }}
/>
```

This allows you to translate app names.

Waiting for your feedbacks. :)

---

Linked to issue #120.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +77,8 @@ showLocation({
     dialogTitle: 'This is the dialog Title', // optional (default: 'Open in Maps')
     dialogMessage: 'This is the amazing dialog Message', // optional (default: 'What app would you like to use?')
     cancelText: 'This is the cancel button text', // optional (default: 'Cancel')
-    appsWhiteList: ['google-maps'] // optionally you can set which apps to show (default: will show all supported apps installed on device)
+    appsWhiteList: ['google-maps'], // optionally you can set which apps to show (default: will show all supported apps installed on device)
+    appTitles: { 'google-maps': 'My custom Google Maps title' } // optionally you can override default app titles
```
Comment (tschoffelen) [inline README.md]: Should we maybe leave this an empty object by default, like we do in the popup example? Just to make sure that people don't simply copy and paste this into their code and have a weird 'My custom Google Maps title' pop up.

Comment (tschoffelen) [reply README.md]: Oh! Or commenting it out, like in the line below it. That might be better.

File: src/constants.js [hunk 2]
```diff
@@ -28,18 +28,20 @@ export function prefixForGoogleMaps (alwaysIncludeGoogle) {
...
-  uber: 'Uber',
-  lyft: 'Lyft',
-  transit: 'The Transit App',
-  waze: 'Waze',
-  yandex: 'Yandex.Navi',
-  moovit: 'Moovit',
-  'yandex-maps': 'Yandex Maps',
-  kakaomap: 'Kakao Maps'
+export function generateTitles (titles) {
+  return Object.assign({
```
Comment (tschoffelen) [inline src/constants.js]: Should we maybe use ES6 syntax here? Something like:

```js
{
    'one': 'value',
    'other': 'value',
    ...(titles || {})
}
```

File: README.md (hunk 1 above)
Comment (exiam) [reply README.md]: OK, I understand. I have commented this line.

File: src/constants.js (hunk 2 above)
Comment (exiam) [reply src/constants.js]: OK, fixed. :)

**Comment by exiam** [comment]
I removed some semicolons or double quote I left behind, according to the global syntax.

**Comment by tschoffelen** [comment]
Released to NPM as `react-native-map-link@2.7.1`

Thanks again for contributing!

**Pull request merged**